src/configurePipeline/client/azureResourceClient.ts

```diff
@@ -0,0 +1,39 @@
+import * as ResourceManagementClient from 'azure-arm-resource/lib/resource/resourceManagementClient';
+import { ServiceClientCredentials } from 'ms-rest';
+import { ResourceListResult, GenericResource } from 'azure-arm-resource/lib/resource/models';
+
+export class AzureResourceClient {
+    private azureRmClient: ResourceManagementClient;
+
+    constructor(credentials: ServiceClientCredentials, subscriptionId: string) {
+        this.azureRmClient = new ResourceManagementClient(credentials, subscriptionId);
+    }
+
+    public async getResourceList(resourceProvider: string) {
+        let resourceListResult: ResourceListResult = await this.azureRmClient.resources.list({ filter: "$resource.Type eq " + resourceProvider });
+
+        let resourceList: Array<AzureResource>;
+        resourceListResult.forEach((resource) => {
+            resourceList.push();
+        });
+
+    }
+}
+
+export class AzureResource {
+    public resourceId: string;
+    public resourceName: string;
+    public resourceGroupName: string;
+    public subscriptionId: string;
+    public isFunctional: boolean;
```

Comment:
what is isFunctional property ?

Reply:
This will be used for determining if the resource is in usable state, which some resources even after being listed are not. like service fabric, aks, vmss etc.
Should we rename it to isActive ? or something in that sort ?

Reply:
Removed the object